jetty-server/src/main/java/org/eclipse/jetty/server/CachedContentFactory.java

```diff
@@ -589,7 +601,7 @@ public ByteBuffer getDirectBuffer()
                     else
                         buffer = _mappedBuffer.get();
                 }
-                else
+                else if (_resource.length() < _maxCachedFileSize)
```

Comment:
Need a comment here explaining this condition... and why it doesn't apply to mapped buffers above

Reply:
Done. fixed in commit 602c44ec96966b0a40b3b3fae0e465a2a114ca1f